Move hello outside of the main installation

sampaccoud opened the pull request:
## Purpose

Move hello outside of the main installation.

## Proposal

Move the hello project to a separate example directory. 
The hello project  can be deployed by manually overriding the "templates" and "group_var" directories at the root of the project by the ones provided in this example.

sampaccoud committed:
🔥(projects) limit default project to just hello

The default project should be as simple as possible.
The openedx and richie installations are saved and
will be added in a future commit in a separate
"examples" folder containing a "templates" and a
"group_var" directories that can be mounted on the
root ones when running a docker container in Arnold.

cchaudier commented on group_vars/all/main.yml:
```diff
@@ -1,114 +1,23 @@
-cms_port:       8081
-lms_port:       8071
-richie_port:    8061
-aliases_port:   8999
```
This line must be in all for me unless if you think some project have no redirection nginx pod ?

sampaccoud replied:
this comment and the following are not relevant as I am removing these files.

cchaudier commented on group_vars/all/main.yml:
```diff
@@ -1,114 +1,23 @@
-  - openshift/edxapp/dc/lms.yml
-  - openshift/edxapp/dc/memcached.yml
-  - openshift/common/dc/elasticsearch.yml
-  - openshift/common/dc/nginx.yml
```
DRY : Nginx is on all project no ?

cchaudier commented on group_vars/all/main.yml:
```diff
@@ -1,114 +1,23 @@
-  - openshift/common/svc/elasticsearch.yml
-  - openshift/common/svc/mongodb.yml
-  - openshift/common/svc/mysql.yml
-  - openshift/common/svc/nginx.yml
```
DRY : Nginx is on all project no ?

cchaudier commented on group_vars/all/openshift_routes.yml:
```diff
@@ -1,16 +1,14 @@
 ---
-# URL
+# Url
```
URL is an acronym. I think it must stay in upercasse.

sampaccoud replied:
url is always mentioned in low cap in code ;-)

cchaudier commented on templates/openshift/hello/dc/hello.yml.j2:
```diff
@@ -6,7 +6,7 @@ metadata:
   namespace: "{{ project_name }}"
 spec:
-  replicas: 1  # number of pods we want
+  replicas: 2  # number of pods we want
```
why 2 replicas ? Is for testing de blue/green switch ? the load-balancer ?

sampaccoud replied:
yes load balancer and why not.

cchaudier reviewed (commented):
It's the bigbang ! 
I've some comments and questions.

sampaccoud committed:
✨(hello) make Hello an example project

We add a new directory "examples". In this directory
we can add example projects that can override the
"templates" and/or the "groups_vars" directories. This
is done by mounting them on the directories with the
same name at the root of the project when running
a docker command.

cchaudier commented on examples/hello/group_vars/secret/patient0/development/edxapp/credentials.vault.yml:
```diff
@@ -0,0 +1,42 @@
+$ANSIBLE_VAULT;1.1;AES256
```
An exemple credentials need to be vaulted ?
Are you sur that this file is linked to the hello exemple ?

sampaccoud replied:
You're right, there's no secrets files required for hello. Removed them all.

cchaudier reviewed (commented):
Just one question about credentials and exemple.

jmaupetit reviewed (commented):
Can you document the usage of this hello example?

sampaccoud commented:
@jmaupetit well, not really for the moment as it will depend on your refactoring. For the moment I was just taking it out of the main project as it has nothing to do with it.

cchaudier commented on examples/hello/group_vars/secret/patient0/development/richie/credentials.vault.yml:
```diff
@@ -0,0 +1,24 @@
+$ANSIBLE_VAULT;1.1;AES256
```
Are you sur that this file is linked to the hello exemple ?

sampaccoud replied:
:+1: removed it

cchaudier reviewed (commented):
Juste some comments

jmaupetit commented:
Ok, I see. I've declared an issue for that (see #46).

jmaupetit reviewed (approved):
Nice move.

cchaudier commented:
Great job !
